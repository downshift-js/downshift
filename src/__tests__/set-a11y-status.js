jest.useFakeTimers()

beforeEach(() => {
  document.body.innerHTML = ''
})

test('sets the status', () => {
  const setA11yStatus = setup()
  setA11yStatus('hello')
  expect(document.body.firstChild).toMatchSnapshot()
})

test('replaces the status with a different one', () => {
  const setA11yStatus = setup()
  setA11yStatus('hello')
  setA11yStatus('goodbye')
  expect(document.body.firstChild).toMatchSnapshot()
})

test('does add anything for an empty string', () => {
  const setA11yStatus = setup()
  setA11yStatus('')
  expect(document.body.firstChild).toMatchSnapshot()
})

test('escapes HTML', () => {
  const setA11yStatus = setup()
  setA11yStatus('<script>alert("!!!")</script>')
  expect(document.body.firstChild).toMatchSnapshot()
})

test('performs cleanup after a timeout', () => {
  const setA11yStatus = setup()
  setA11yStatus('hello')
  jest.runAllTimers()
  expect(document.body.firstChild).toMatchSnapshot()
})

function setup() {
  jest.resetModules()
  return require('../set-a11y-status').default
}
