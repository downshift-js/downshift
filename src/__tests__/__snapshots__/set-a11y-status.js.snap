// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does add anything for an empty string 1`] = `
<div
  aria-live="polite"
  aria-relevant="additions text"
  id="a11y-status-message"
  role="status"
  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
/>
`;

exports[`escapes HTML 1`] = `
<div
  aria-live="polite"
  aria-relevant="additions text"
  id="a11y-status-message"
  role="status"
  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
>
  &lt;script&gt;alert("!!!")&lt;/script&gt;
</div>
`;

exports[`performs cleanup after a timeout 1`] = `
<div
  aria-live="polite"
  aria-relevant="additions text"
  id="a11y-status-message"
  role="status"
  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
/>
`;

exports[`replaces the status with a different one 1`] = `
<div
  aria-live="polite"
  aria-relevant="additions text"
  id="a11y-status-message"
  role="status"
  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
>
  goodbye
</div>
`;

exports[`sets the status 1`] = `
<div
  aria-live="polite"
  aria-relevant="additions text"
  id="a11y-status-message"
  role="status"
  style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
>
  hello
</div>
`;
